iree/turbine/kernel/wave/promotion.py

```diff
@@ -122,6 +122,11 @@ def promote_node(
     symbolic_shape = node.type.symbolic_shape
     with node.graph.inserting_after(node.graph._root):
         constrained_shape = get_constrained_shape(symbolic_shape, constraints)
+        # If the read/write involves shared memory, then use the distributed shape
```

Comment:
read/write here would most likely involves shared memory. Should detail that if already has existing shared memory with pre-specified distributed_shape (e.g defined from kernel level), then we use the distribtued shape if not we use constrained/block shape deduced from constraint